lib/samples/style_graphics_with_renderer/style_graphics_with_renderer.dart

```diff
@@ -0,0 +1,291 @@
+    // Create a graphics overlay for the ellipse.
+    final ellipseGraphicsOverlay = GraphicsOverlay();
+    // Create and assign a simple renderer for the ellipse.
+    ellipseGraphicsOverlay.renderer = SimpleRenderer(symbol: ellipseFillSymbol);
+
+    // Create an ellipse graphic.
+    final ellipseCenter = ArcGISPoint(
+      x: 40e5,
+      y: 25e5,
+      spatialReference: SpatialReference.webMercator,
+    );
+    final parameters = GeodesicEllipseParameters(
+      axisDirection: -45,
+      angularUnit: AngularUnit(unitId: AngularUnitId.degrees),
+      center: ellipseCenter,
+      linearUnit: LinearUnit(unitId: LinearUnitId.kilometers),
+      maxPointCount: 100,
+      maxSegmentLength: 20,
+      geometryType: GeometryType.polygon,
+      semiAxis1Length: 200,
+      semiAxis2Length: 400,
+    );
+    final ellipseGeometry =
+        GeometryEngine.ellipseGeodesic(parameters: parameters);
+    final ellipseGraphic = Graphic(geometry: ellipseGeometry);
+
+    // Add the graphic to the overlay.
+    ellipseGraphicsOverlay.graphics.add(ellipseGraphic);
+    return ellipseGraphicsOverlay;
+  }
+
+  GraphicsOverlay getCurvedPolygonGraphicsOverlay() {
+    // Create a simple fill symbol with outline.
+    final curvedLineSymbol = SimpleLineSymbol(color: Colors.black);
+    final curvedFillSymbol = SimpleFillSymbol(
+      color: Colors.red,
+      outline: curvedLineSymbol,
+    );
+    // Create a graphics overlay for the polygons with curve segments.
+    final curvedGraphicsOverlay = GraphicsOverlay();
+    // Create and assign a simple renderer to the graphics overlay.
+    curvedGraphicsOverlay.renderer = SimpleRenderer(symbol: curvedFillSymbol);
+
+    // Create a heart-shape graphic from ArcGISSegment's.
+    final origin = ArcGISPoint(
+      x: 40e5,
+      y: 5e5,
+      spatialReference: SpatialReference.webMercator,
+    );
+    final heartGeometry = getHeartGeometry(center: origin, sideLength: 10e5);
+    final heartGraphic = Graphic(geometry: heartGeometry);
+
+    // Add the graphic to the overlay.
+    curvedGraphicsOverlay.graphics.add(heartGraphic);
+    return curvedGraphicsOverlay;
+  }
+
+  Geometry? getHeartGeometry({
+    required ArcGISPoint center,
+    required double sideLength,
+  }) {
+    // The side length should be always greater than 0.
+    if (sideLength <= 0) {
+      return null;
+    }
+
+    final spatialReference = center.spatialReference;
+    // The x and y coordinates to simplify the calculation.
+    final minX = center.x - 0.5 * sideLength;
+    final minY = center.y - 0.5 * sideLength;
+    // The radius of the arcs.
+    final arcRadius = sideLength * 0.25;
+
+    // Bottom left curve.
+    final leftCurveStart =
+        ArcGISPoint(x: center.x, y: minY, spatialReference: spatialReference);
+    final leftCurveEnd = ArcGISPoint(
+        x: minX,
+        y: minY + 0.75 * sideLength,
+        spatialReference: spatialReference);
```

Comment:
I think `dart analyze` would tell you to put a trailing comma at the end of this, and then `dart format` would change the indentation to the desired outcome.

Reply:
and a couple more cases below

Reply:
Sorry, I am going to set that up on every save now.